### PR TITLE
Finish filter words with javascript

### PR DESCRIPTION
**Pull request by thiennv57**

### CODE REVIEW

File: app/controllers/words_controller.rb [hunk 1]
```diff
@@ -3,7 +3,11 @@ class WordsController < ApplicationController
 
   def index
     @categories = Category.all
-    @words = Word.in_category(params[:category_id]).paginate page: params[:page],
-      per_page: Settings.paginate_per_page
+    if params[:filter_state]
+      @words = Word.in_category(params[:category_id]).send(params[:filter_state], current_user)
+        .paginate page: params[:page], per_page: Settings.paginate_per_page
+    else
+      @words = Word.all
```
Comment (ThuBM) [inline app/controllers/words_controller.rb]: paginate?

File: app/models/word.rb [hunk 2]
```diff
@@ -12,6 +12,7 @@ class Word < ActiveRecord::Base
   scope :not_learned, ->user{where("id NOT IN (
     SELECT word_id FROM results WHERE lesson_id IN ( SELECT id FROM lessons WHERE user_id = ?
```
Comment (ThuBM) [inline app/models/word.rb]: indent

File: app/views/shared/_filter_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<%= form_for words_path, method: :get, remote: true do |f| %>
+  <div class="row">
+    <div class="col-md-6 col-md-offset-2">
+      <%= select_tag "category_id", options_from_collection_for_select(@categories, "id", "name", 
+        params[:category_id]), include_blank: t("word.all"), class: "form-control"%>
```
Comment (ThuBM) [inline app/views/shared/_filter_form.html.erb]: add space before %>

**Comment by ThuBM** [comment]
Ok

File: app/views/words/_word.html.erb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<table class="table table-hover">
+  <tr class="success">
+    <th><%= t "word" %></th>
+    <th><%= t "meaning" %></th>
+    <% @words.each do |w| %>
```
Comment (chuanhtuan) [inline app/views/words/_word.html.erb]: <% @words.each do |word| %>

File: app/controllers/words_controller.rb [hunk 5]
```diff
@@ -3,7 +3,12 @@ class WordsController < ApplicationController
...
```
Comment (dieunb) [inline app/controllers/words_controller.rb]: remove `all`

File: app/views/shared/_filter_form.html.erb [hunk 6]
```diff
@@ -0,0 +1,21 @@
+<%= form_for words_path, method: :get, remote: true do |f| %>
+  <div class="row">
+    <div class="col-md-6 col-md-offset-2">
+      <%= select_tag "category_id", options_from_collection_for_select(@categories, "id", "name", 
+        params[:category_id]), include_blank: t("word.all"), class: "form-control" %>
+      <% [Settings.filter.learned, Settings.filter.not_learned, Settings.filter.get_all].each do |value| %>
```
Comment (dieunb) [inline app/views/shared/_filter_form.html.erb]: `Settings.filters.each do ...`

File: app/views/categories/.fuse_hidden0000165000000002 [hunk 7]
```diff
@@ -0,0 +1,18 @@
+<div class="row">
```
Comment (dieunb) [inline app/views/categories/.fuse_hidden0000165000000002]: remove this file

**Pull request merged**